test/adapter/json_api/relationship_test.rb

```diff
@@ -4,40 +4,36 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class RelationshipTest < ActiveSupport::TestCase
-        setup do
-          @blog = Blog.new(id: 1)
-          @author = Author.new(id: 1, name: 'Steve K.', blog: @blog)
-          @serializer = BlogSerializer.new(@blog)
-          ActionController::Base.cache_store.clear
-        end
-
         def test_relationship_with_data
           expected = {
             data: {
               id: '1',
               type: 'blogs'
             }
           }
-          test_relationship(expected, options: { include_data: true })
+
+          model_attributes = { blog: Blog.new(id: 1) }
+          relationship_name = :blog
+          model = new_model(model_attributes)
+          actual = build_serializer_and_serialize_relationship(model, relationship_name) do
+            has_one :blog
+          end
+          assert_equal(expected, actual)
         end
 
         def test_relationship_with_nil_model
-          @serializer = BlogSerializer.new(nil)
           expected = { data: nil }
-          test_relationship(expected, options: { include_data: true })
-        end
 
-        def test_relationship_with_nil_serializer
```

Comment:
a relationship with a nil serializer that isn't using virtual values (i.e. doesn't need a serializer in the firsts place) isn't a supported use case